benchto-driver/src/main/java/com/teradata/benchto/driver/listeners/measurements/ThroughputMeasurementProvider.java

```diff
@@ -29,12 +30,17 @@
         implements PostExecutionMeasurementProvider
 {
     @Override
-    public List<Measurement> loadMeasurements(Measurable measurable)
+    public CompletableFuture<List<Measurement>> loadMeasurements(Measurable measurable)
     {
+        List<Measurement> measurements;
         if (measurable instanceof BenchmarkExecutionResult && measurable.getBenchmark().isConcurrent() && measurable.isSuccessful()) {
-            return ImmutableList.of(measurement("throughput", "QUERY_PER_SECOND", calculateThroughput((BenchmarkExecutionResult) measurable)));
+            measurements = ImmutableList.of(measurement("throughput", "QUERY_PER_SECOND", calculateThroughput((BenchmarkExecutionResult) measurable)));
         }
-        return emptyList();
+        else {
+            measurements = emptyList();
+        }
+
+        return CompletableFuture.completedFuture(measurements);
```

Comment:
This is "Allow PostExecutionMeasurementProviders to be async" commit -- it does not introduce any asynchrony, it just makes API async-ready (by future-izing them). Asynchrony comes with "Query Graphite after configurable delay" commit and only for Graphite metrics loader (others don't need it).